pypeit/core/skysub.py

```diff
@@ -116,6 +116,9 @@ def global_skysub(image, ivar, tilts, thismask, slit_left, slit_righ, inmask=Non
     piximg = tilts * (nspec-1)
     if inmask is None:
         inmask = (ivar > 0.0) & thismask & np.isfinite(image) & np.isfinite(ivar)
+    else:
```

Comment:
The docs require that the input be of type bool. Rather than casting, I would prefer that you just fault if this is not a boolean mask. If necessary you can cast outside of this routine. All the codes in PypeIt use boolean arrays as masks, and we are not providing this flexibility to use 1s or 0s in arrays elsewhere.